vimscript.html.markdown

```diff
@@ -0,0 +1,572 @@
+echo  a:                |" All function arguments
+echo  v:                |" All Vim variables
+
+" Constant variables
+const x = 10            |" See :h :const, :h :lockvar
+
+" Function reference variables have the same restrictions as function names
+let IsString = {x -> type(x) == type('')}    |" Global: capital letter
+let s:isNumber = {x -> type(x) == type(0)}   |" Local: any name allowed
+```
+
+When omitted the scope `g:` is implied, except in functions, there `l:` is
+implied.
+
+### Multiple value binding (list unpacking)
+
+```vim
+" Assign values of list to multiple variables (number of items must match)
+let [x, y] = [1, 2]
+
+" Assign the remainer to a rest variable (note the semicolon)
+let [mother, father; children] = ['Alice', 'Bob', 'Carol', 'Dennis', 'Emily']
+```
+
+## Flow control
+
+### Conditional
+
+Conditions are set between `if` and `endif`. They can be nested.
+
+```vim
+if condition
+	  echo 'First condition'
+elseif another_condition
+	  echo 'Second condition'
+else
+	  echo 'Fail'
+endif
+```
+
+### Loops
+
+Two types of loops: `:for` and `:while`. Use `:continue` to skip to the next
+iteration, `:break` to break out of the loop.
+
+#### For-loop
+
+For-loops iterate over lists and nothing else. If you want to iterate over
+another sequence you need to use a function which will create a list.
+
+```vim
+" Iterate over a list
+for person in ['Alice', 'Bob', 'Carol', 'Dennis', 'Emily']
+		echo 'Hello ' . person
+endfor
+
+" Iterate over a nested list by unpacking it
+for [x, y] in [[1, 0], [0, 1], [-1, 0], [0, -1]]
+		echo 'Position: x ='  . x . ', y = ' . y
+endfor
+
+" Iterate over a range of numbers
+for i in range(10, 0, -1)  " Count down from 10
+    echo 'T minus'  . i
+endfor
+
+" Iterate over the keys of a dictionary
+for symbol in keys({'π': 3.14, 'e': 2.71})
+		echo 'The constant ' . symbol . ' is a transcendent number'
+endfor
+
+" Iterate over the values of a dictionary
+for value in keys({'π': 3.14, 'e': 2.71})
+		echo 'The value ' . symbol . ' approximates a transcendent number'
+endfor
+
+" Iterate over the keys and values of a dictionary
+for [symbol, value] in items({'π': 3.14, 'e': 2.71})
+		echo 'The number ' . symbol . ' is approximately ' . value
+endfor
+```
+
+#### While-loops
+
+```vim
+while !there_yet
+		echo 'Are we there yet?'
+endwhile
+```
+
+### Exception handling
+See `:h exception-handling`
+
+Throw new exceptions as strings, catch them by pattern-matching a regular
+expression against the string
+
+```vim
+" Throw new exception
+throw "Wrong arguments"
+
+" Guard against an exception (the second catch matches any exception)
+try
+		source path/to/file
+catch /Cannot open/
+		echo 'Looks like that file does not exist'
+catch /.*/
+    echo 'Something went wrong, but I don't know what'
+finally
+		echo 'I'm done trying'
+endtry
+```
+
+
+## Functions
+
+### Defining functions
+
+```vim
+" Function names have to start with a capital letter
```

Comment:
```suggestion
" Unscoped function names have to start with a capital letter
```

Reply:
Perhaps unnecessary since this is made clear below with `s:addNumersLoudly(x, y)`?

Reply:
Good point.